From a5ff0a2261e2df7b2157e6509cdbecf65033e578 Mon Sep 17 00:00:00 2001
From: David Hanina <dhanina@redhat.com>
Date: Tue, 10 Mar 2026 10:26:33 +0100
Subject: [PATCH] Allow 32bit gid

We should allow 32bit groups, by setting maxvalue we allow that.

Fixes: https://pagure.io/freeipa/issue/9953
Signed-off-by: David Hanina <dhanina@redhat.com>
Reviewed-By: David Hanina <dhanina@redhat.com>
Reviewed-By: Sudhir Menon <sumenon@redhat.com>
---
 ipaserver/plugins/group.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ipaserver/plugins/group.py b/ipaserver/plugins/group.py
index f05a39f69ec3eb3257909bd61b42a9a21212c14d..308e5458c1d00be34753f42675367a7307331514 100644
--- a/ipaserver/plugins/group.py
+++ b/ipaserver/plugins/group.py
@@ -26,6 +26,7 @@ import re
 from ipalib import api
 from ipalib import Int, Str, Flag
 from ipalib.constants import PATTERN_GROUPUSER_NAME, ERRMSG_GROUPUSER_NAME
+from ipalib.parameters import MAX_UINT32
 from ipalib.plugable import Registry
 from .baseldap import (
     add_external_post_callback,
@@ -354,6 +355,7 @@ class group(LDAPObject):
             label=_('GID'),
             doc=_('GID (use this option to set it manually)'),
             minvalue=1,
+            maxvalue=MAX_UINT32,
         ),
         ipaexternalmember_param,
     )
-- 
2.52.0

